Accept/decline pending users

AnsonLee99 opened the pull request:
### Administrative Info
Monday Board ID: 1181559666 
Make sure your branch name conforms to: `<feature/staging/hotfix/...>/[username]/[Monday Item ID]-[3-4 word description separated by dashes]`. Otherwise, please rename your branch and create a new PR.

### Changes
What changes did you make?
1) Added a GET route at `/api/users/:user_id` to retrieve a specific user with the given user_id
2) Created page and process for pending users to be either accepted or declined. 

### Testing
Accepted and/or declined various test accounts and verified changes in MongoDB

### Confirmation of Change 
Page to accept or decline a pending user:
![image](https://user-images.githubusercontent.com/36427506/119591714-fac12d80-bd8b-11eb-94b2-052ee4fa3a46.png)

Popup and dropdown for accepting a pending user ("Done" button is disabled until a role is selected"):
![image](https://user-images.githubusercontent.com/36427506/119591767-13314800-bd8c-11eb-99a3-bed199e3fdca.png)

Popup for declining a pending user:
![image](https://user-images.githubusercontent.com/36427506/119591816-2512eb00-bd8c-11eb-9983-4e8c3adc6db8.png)

AnsonLee99 committed:
Accept/decline pending users

hahsu commented:
> It seems a bit strange to be able to assign the Pending role back to a user. Can we remove that role from the dropdown?

To add onto this, since we only recruit for Designers and Developers outside of TSE, the dropdown menu should only have Designers and Developers. Every other role is an internal recruit, so you go to the settings page and update your role (or if you don't have the permissions to do that, a PVP will update your role for you), so all of the other roles are not necessary in this dropdown.

hollyham commented:
> This PR is linked to [Item 1309249030](https://triton-software-engineering.monday.com/boards/1309249030/pulses/1309249030) on [Board 1309249030](https://triton-software-engineering.monday.com/boards/1309249030) and has now been put into Code Review status!

Board ID should be 1181559666

AnsonLee99 commented:
> > This PR is linked to [Item 1309249030](https://triton-software-engineering.monday.com/boards/1309249030/pulses/1309249030) on [Board 1309249030](https://triton-software-engineering.monday.com/boards/1309249030) and has now been put into Code Review status!
> 
> Board ID should be 1181559666 (right now it appears to be set to the task ID).

Ah whoops, my bad!

AnsonLee99 committed:
Merge branch 'main' of https://github.com/TritonSE/TSE-Oktavian into feature/ansonlee99/1309249030-pending-user-approval

AnsonLee99 committed:
added external_recruiment field to roles

AnsonLee99 commented:
Did a couple things:
1) Added `external_recruitment` field to role model
2) Made Developer and Designer roles external_recruitment
3) Fixed the createUser() function in users.js as the registration process was broken

Screenshot of assignable roles when accepting pending user:
![image](https://user-images.githubusercontent.com/36427506/119957676-23e1e980-bf57-11eb-9fa9-27eb2320a202.png)

Screenshot of `external_recruitment` field (the field is false for all the other roles):
![image](https://user-images.githubusercontent.com/36427506/119957809-47a52f80-bf57-11eb-87a4-bf835c775965.png)

lynguyxn reviewed (approved):
lgtm!

hahsu commented on backend/services/users.js:
```diff
@@ -36,7 +36,10 @@ async function createUser(raw_user) {
-  return new User(raw_user_no_secret).save();
+  user = new User(raw_user_no_secret);
+  await user.save();
+  user.role = pending_role;
```
Why is this line need if we have on line 35: `role: pending_role._id`

justinyaodu replied:
This seems like a merge conflict that was improperly resolved. The line being deleted in that diff came from [this PR](https://github.com/TritonSE/TSE-Oktavian/commit/6d2e7929f500044ef893378eb1184330ec46f17b#diff-c8feca3408825c1b0293f25ad5e0a50288093429959e8ec7448e8a5f3837e9bcR39) where I simplified the logic.

justinyaodu replied:
@AnsonLee99, could you please undo this change before merging?

AnsonLee99 replied:
No, this code is needed because the role field needs to be populated for PageContainer to not throw an error after the registration process.. I fixed this in a previous PR but there was a recent commit that broke it again.

hahsu reviewed (approved):
I left one small question as a comment, but otherwise lgtm

justinyaodu commented on backend/services/users.js:
```diff
@@ -36,7 +36,10 @@ async function createUser(raw_user) {
 
   delete raw_user_no_secret.secret;
-  return new User(raw_user_no_secret).save();
+  user = new User(raw_user_no_secret);
+  await user.save();
+  user.role = pending_role;
```
```suggestion
  return new User(raw_user_no_secret).save();
```

AnsonLee99 replied:
this breaks the registration process

AnsonLee99 committed:
Merge branch 'main' into feature/ansonlee99/1309249030-pending-user-approval

justinyaodu reviewed (approved):
LGTM!

AnsonLee99 commented on backend/services/users.js:
```diff
@@ -36,7 +36,10 @@ async function createUser(raw_user) {
-  return new User(raw_user_no_secret).save();
+  user = new User(raw_user_no_secret);
+  await user.save();
+  user.role = pending_role;
```
```suggestion
  user.role = pending_role;       // Populate role field in returned user object
```

AnsonLee99 committed:
Update backend/services/users.js

AnsonLee99 commented on backend/services/users.js:
```diff
@@ -36,7 +36,10 @@ async function createUser(raw_user) {
-  return new User(raw_user_no_secret).save();
+  user = new User(raw_user_no_secret);
+  await user.save();
+  user.role = pending_role;       // Populate role field in returned user object
```
```suggestion
  // Populate role field in returned user object
  user.role = pending_role;       
```

AnsonLee99 committed:
Update backend/services/users.js

AnsonLee99 committed:
add comment for populating role

AnsonLee99 committed:
lint error

AnsonLee99 committed:
Merge branch 'main' into feature/ansonlee99/1309249030-pending-user-approval
